addons/html_editor/static/src/main/link/link_plugin.js

```diff
@@ -33,6 +34,27 @@ function isLinkActive(selection) {
     return false;
 }
 
+/**
+ * @param {EditorSelection} selection
+ */
+function isLinkDisabled(selection) {
+    const blockElements = childNodes(selection.commonAncestorContainer).filter(
+        (el) => isBlock(el) && el.isContentEditable
+    );
+    if (blockElements.length !== 1) {
+        return blockElements.length > 1;
+    }
+    // in case of ctrl+a
+    const { anchorNode, anchorOffset, focusNode, focusOffset } = selection;
+    const [validAnchorNode] = getDeepestPosition(anchorNode, anchorOffset);
+    const [validFocusNode] = getDeepestPosition(focusNode, focusOffset);
+    const commonAncestor = getCommonAncestor([validAnchorNode, validFocusNode]);
+    if (isTextNode(commonAncestor)) {
+        return false;
+    }
```

Comment:
I don't understand this special case. You say "in case of ctrl+a" but that doesn't explain why it requires a special case in the first place. Why Would the `commonAncestor` be a `TextNode` when doing CTRL+A? Did you mean in an empty editable or something?

In any case, this whole function looks quite big just to test if the selection spans a block. I'm sure there are other places in the code where we already make that kind of checks and it's shorter/simpler.